presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -188,7 +188,8 @@ public ListenableFuture<?> isBlocked()
     public boolean needsInput()
     {
         // We can skip probe for empty build input only when probeOnOuterSide is false
-        if (optimizeProbeForEmptyBuild && !probeOnOuterSide) {
+        // When finishing is true, the partition in the lookup source may have been released and set to null, skip if finishing is true.
+        if (optimizeProbeForEmptyBuild && !probeOnOuterSide && !finishing) {
```

Comment:
Is checking for finishing alone not sufficient?

Reply:
It is (and it should be) as we do not attempt to release a hash table if it's not finishing.